### DISCUSSION TITLE
Node locked license with multiple users

### DISCUSSION THREAD

**Discussion by RobWunderlich** [keygen-sh/community#98] [Q&A] [answered]
I'm currently using node locked licenses. I want to extend this model further to limit each license to a set of named userids. I thought of using the KeyGen User object, but a License has only one User. Any suggestions on how/where to model this requirement in KeyGen?

**Comment by ezekg** [comment, marked as answer]
Issue for this is here: https://github.com/keygen-sh/keygen-api/issues/534. Usually, I'd say to use machines to activate individual users as a workaround (fingerprint being the user's email), but since you're wanting to activate machines and assign users at the same time, right now, multiple users per-license isn't possible with that model.

You may be able to add the set of user emails/IDs to the license's metadata, but enforcement of that would fall on you, outside of Keygen.

**Reply by RobWunderlich** [reply]
Thanks, I'll watch that issue.  The size limits on metadata might eliminate metadata as an option. I'm likely to typically have 5-20 userids, but I may need to support edge cases of 100s of users. I'll implement my own License -> Userid store for now.
